ecosystem_tests/__init__.py

```diff
@@ -50,74 +59,242 @@ def flush(self):
+        """initialize or load a local profile.
+        :returns: return the local env
+        :rtype: cloudify.workflows.local._Environment
+        """
+
         cfy_storage = FileStorage()
         cfy_storage.__init__(self.blueprintdir)
         if os.environ.get('ECOSYSTEM_SESSION_LOADED', False):
             return load_env(cfy_storage)
         else:
             cfy_local = init_env(
-                blueprint_path,
-                inputs=self.inputs(),
+                os.path.join(
+                    self.blueprint_dir,
+                    self.blueprint_archive,
+                    self.blueprint_file_name),
+                inputs=self.inputs,
                 storage=cfy_storage,
                 ignored_modules=IGNORED_LOCAL_WORKFLOW_MODULES)
             os.environ['ECOSYSTEM_SESSION_LOADED'] = 'true'
             return cfy_local
 
-    def setUp(self,
-              blueprint_file_name,
-              blueprint_archive='cloudify-environment-setup-latest',
-              plugins_to_upload=None,
-              package_url=None,
-              sensitive_data=None):
-
-        if 'ECOSYSTEM_SESSION_PASSWORD' not in os.environ:
-            os.environ['ECOSYSTEM_SESSION_PASSWORD'] = create_password()
-        self.password = os.environ['ECOSYSTEM_SESSION_PASSWORD']
-        self.sensitive_data = sensitive_data or []
-        self.sensitive_data.append(self.password)
-        sys.stdout = PasswordFilter(self.sensitive_data, sys.stdout)
-        sys.stderr = PasswordFilter(self.sensitive_data, sys.stderr)
-        super(TestLocal, self).setUp()
-        self.blueprinturl = 'https://github.com/cloudify-examples/' \
-                            'cloudify-environment-setup/archive/latest.zip'
-        if 'ECOSYSTEM_SESSION_BLUEPRINT_DIR' not in os.environ:
-            os.environ['ECOSYSTEM_SESSION_BLUEPRINT_DIR'] = tempfile.mkdtemp()
-        self.blueprintdir = os.environ['ECOSYSTEM_SESSION_BLUEPRINT_DIR']
-        self.blueprint_zip = os.path.join(self.blueprintdir, 'blueprint.zip')
-        self.blueprint_file_name = blueprint_file_name
-        self.blueprint_archive = blueprint_archive
-        self.blueprint_path = \
-            os.path.join(
-                self.blueprintdir,
-                self.blueprint_archive,
-                self.blueprint_file_name)
-        self.package_url = package_url
-        self.plugins_to_upload = \
-            plugins_to_upload or [(DIAMOND_WAGON, DIAMOND_YAML)]
-        self.cfy_local = self.setup_cfy_local()
-
     def install_manager(self):
+        """install a cloudify manager using local profile
+
+        :returns: nothing
+        :rtype: NoneType
+        """
+
         self.addCleanup(self.uninstall_manager)
         self.cfy_local.execute(
             'install',
             task_retries=45,
             task_retry_interval=10)
         self.node_instances = \
             self.cfy_local.storage.get_node_instances()
+        self.manager_ip = get_manager_ip(
```

Comment:
@EarthmanT I'm not sure if we can generalized this, because for example in order to get manager public ip on GCP you need to get it using this 
```
instance.runtime_properties['networkInterfaces'][0][
                    'accessConfigs'][0]['natIP']
```

Reply:
I think maybe we should make `get_manager_ip` a method in the class (instead of a util). Then we can just override it for certain plugins.
 What do you think?

Reply:
Yeah, this will be a good